### ISSUE TITLE
[libc] Implement `capget` and `capset` syscall wrappers 

### ISSUE THREAD

**Issue by SchrodingerZhu** [llvm/llvm-project#80037]
Implement https://man7.org/linux/man-pages/man2/capset.2.html wrappers. The manpage mentioned that it was not provided by `glibc` but MUSL implement them as an extension:
https://github.com/bminor/musl/blob/f314e133929b6379eccc632bef32eaebb66a7335/src/linux/cap.c#L8

This is helpful in #79645.

**Comment by vinayakdsci** [comment]
@SchrodingerZhu hello! Could you give me a few pointers on where the code has to be placed? I can implement the wrappers, but I am unable to judge where to place them.

**Comment by SchrodingerZhu** [comment]
sys/capability.h may be a good place to good. I am little bit concerned that we may conflict with libcap. @nickdesaulniers

**Comment by nickdesaulniers** [comment]
ah, that man page says:

> The portable interfaces are cap_set_proc(3) and cap_get_proc(3);
       if possible, you should use those interfaces in applications;

Perhaps it would be better to implement those instead.

**Comment by vinayakdsci** [comment]
I have a small doubt here. Neither glibc nor musl implement `cap_set_proc()` and `cap_get_proc()`, and musl only implements `capget()` and `capset()` just because of the simplicity of their implementation. As libcap already implements all of the capability functionality, would this not cause conflicts? One can simply link to libcap.

@SchrodingerZhu @nickdesaulniers?

**Comment by nickdesaulniers** [comment]
Yeah, the more I look into these APIs, the less interested I am in having implementations for them.

Perhaps we can instead rethink the use of `SYS_capget` in libc/test/src/sys/mman/linux/mlock_test.cpp ?

**Comment by SchrodingerZhu** [comment]
@nickdesaulniers capget may be useful to have. For example, `sched` related tests may also want to check capabilities.

**Comment by kaladron** [comment]
Looking at this, we're likely to get an internal version of this as part of finishing up the syscalls rework, however, we dont' want to expose this and extend the surface of the library for this.  (It's unspecified, not in most other libcs and could easily conflict with a common library symbol).  I'm going to close this issue for now.  (Feel free to reopen if this is the wrong decision)
